### PR TITLE
Make NVCC pass feature tests inside and outside Kokkos build

### PR DESCRIPTION
**Pull request by jjwilke**
Don't rely on `KOKKOS_CXX_STANDARD` variable. Just revert flags if receiving `c++1y` and `c++1z` to `c++14`

### CODE REVIEW

**Comment by masterleinad** [comment]
I can confirm that this fixes #2469.

File: bin/nvcc_wrapper [hunk 1]
```diff
@@ -196,30 +197,28 @@ do
...
-         echo "nvcc_wrapper does not accept intermediate standard flag $1, but I will use -std=c++${KOKKOS_CMAKE_CXX_STANDARD} instead"
-         shared_args="$shared_args -std=c++${KOKKOS_CMAKE_CXX_STANDARD}"
-       else
-	echo "nvcc_wrapper has been given intermediate standard flag $1 - nvcc only accepts full standard flags like -std=c++11"
-	exit 1
-       fi
+       # this is hopefully just occurring in a downstream project during CMake feature tests
+       # we really have no choice here but to accept the flag and "promote" to the corresponding C++ standard
+       echo "nvcc_wrapper does not accept intermediate standard flag $1, but I will use -std=c++14 instead. It is undefined behavior to use this flag. This should only be occurring during CMake configuration."
+       shared_args="$shared_args -std=c++14"
```
Comment (dalg24) [inline bin/nvcc_wrapper]: Why aren't we emitting that warning unconditionally (regardless of whether multiple cxx standard flags are passed or not)?

File: cmake/KokkosConfig.cmake.in [hunk 2]
```diff
@@ -13,6 +13,7 @@ SET(Kokkos_LIBRARIES       @KOKKOS_EXT_LIBRARIES@)
 SET(Kokkos_DEVICES                      @KOKKOS_ENABLED_DEVICES@)
 SET(Kokkos_OPTIONS                      @KOKKOS_ENABLED_OPTIONS@)
 SET(Kokkos_ARCH                         @KOKKOS_ENABLED_ARCH_LIST@)
+SET(Kokkos_CXX_STANDARD                 @KOKKOS_CXX_STANDARD@)
```
Comment (dalg24) [inline cmake/KokkosConfig.cmake.in]: Why did you add this?

File: bin/nvcc_wrapper (hunk 1 above)
Comment (jjwilke) [reply bin/nvcc_wrapper]: The second flag is ignored. But you would prefer that we warn (or error) when an ignored flag is also invalid?

File: cmake/KokkosConfig.cmake.in (hunk 2 above)
Comment (jjwilke) [reply cmake/KokkosConfig.cmake.in]: I had intentions for doing user-friendly errors check downstream when Kokkos loads that needed this variable. I will remove this for now - but this may be a useful variable to export.

File: bin/nvcc_wrapper [hunk 3]
```diff
@@ -102,6 +99,10 @@ depfile_target_arg=""
 # Option to remove duplicate libraries and object files
 remove_duplicate_link_files=0
 
+function warn_std_flag() {
+  echo "nvcc_wrapper - *warning* you have set multiple optimization flags (-std=c++1* or --std=c++1*), only the first is used because nvcc can only accept a single std setting"
+}
```
[the remark this answers is not in the document]
Comment (jjwilke) [reply bin/nvcc_wrapper]: I need that logic in each of the cases so I know whether to execute the `else` block. It's not just a matter of printing the warning or not.  That being said - if you know some bash foo that makes this cleaner, I'm happy to change it.

**Comment by jjwilke** [comment]
This is probably as much as we want to change for 3.0.

I believe the right way to do this for 3.1 is to have `nvcc_wrapper` be different when installed. Have an `nvcc_wrapper.in` that during configure writes the underlying gcc, nvcc, and valid flags/features into the compiler script. All downstream projects MUST use an installed `nvcc_wrapper`. 

The way this works is then:
1. If building Kokkos, Kokkos sets environment variables during CMake configure time that tells the `nvcc_wrapper` to accept and translate certain flags
2. If building downstream, an installed `nvcc_wrapper` has the information on what flags to accept and translate embedded into it.

**Comment by crtrott** [comment]
I don't like the idea of installing a different nvcc_wrapper. Generally I would love to get rid of nvcc_wrapper at some point, but until then I want it to be the minimal logic to fix deficiencies in NVCC and its interaction with CMake.

File: cmake/KokkosConfig.cmake.in (hunk 2 above)
Comment (dalg24) [reply cmake/KokkosConfig.cmake.in]: Drop that line and you've got my approval

Comment (jjwilke) [reply cmake/KokkosConfig.cmake.in]: Removed. Squashed onto previous commit.
